File: vdp-c-sharp/vdp-c-sharp/VDPApiTest/vta/ManageCommunitiesTest.cs
﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Configuration;
using System.Collections.Generic;

namespace Vdp
{
    [TestClass]
    public class ManageCommunitiesTest
    {
        private AbstractVisaAPIClient abstractVisaAPIClient;

        public ManageCommunitiesTest()
        {
            abstractVisaAPIClient = new AbstractVisaAPIClient();
        }

        [TestMethod]
        public void TestGetCommunities()
        {
            string baseUri = "vta/";
            string resourcePath = "v3/communities";
            Dictionary<string, string> headers = new Dictionary<string, string>();
            headers.Add("ServiceId", ConfigurationManager.AppSettings["vtaServiceId"]);
            string status = abstractVisaAPIClient.DoMutualAuthCall(baseUri + resourcePath, "GET", "Get Communities Test", "", headers);
            Assert.AreEqual(status, "OK");
        }
    }
}

File: vdp-ruby/test/vta/manage_communities_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/abstract_visa_api_client', __FILE__)

class ManageCommunitiesTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @abstract_visa_api_client = AbstractVisaAPIClient.new
  end
  
  def test_getCommunities
    base_uri = 'vta/'
    resource_path = 'v3/communities'
    response_code = @abstract_visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Get Communities Test", "get", '', {'ServiceId' => @config['vtaServiceId']})
    assert_equal("200", response_code, "Get Communities test failed")
  end
end